### PR TITLE
#162 Create Validation Statement

### LINKED ISSUE #162: Backend - Abstract int validation
### Description

we validate ints in all the route file post endpoints. however, it's repetitive because we do stuff like `isInt({min: 0}).not().isString()` every time. Make an abstraction for this.

### Background

.

### Proposed Solution

1. make a file in backend utils called validation.utils.ts or something
2. export a const intMinZero function which chains the validation calls for strings mentioned above
3. replace all the existing calls with this new function (don't replace it if it isn't a min 0 int though!)

### Mocks

_No response_

### PR DESCRIPTION
**Pull request by Peyton-McKee**
## Changes

Created validations.utils to house intMinZero function 

Created intMinZero function to replace .isInt({ min: 0 }).not().isString() that covers much of the routing files 

## Notes

No extra notes

## Test Cases

Case 1: Success
Sending a POST request to the /risks/create endpoint with:

{
    "projectId" : 2,
    "createdById": 2,
    "detail" : "test"
}

The result should be:
Status 200: OK

{
    "message": "Successfully created risk #b88ca09a-2475-4087-bddf-768b5d7b1e19."
}
and create new risk with specified inputs

Case 2: Fail: Invalid Value for projectId (i.e. <0)

Sending a POST request without the psId field:

{
    "projectId" : -1,
    "createdById": 2,
    "detail" : "test"
}
The result should be:
Status 400: Bad Request

{
"errors":[{"value":-1,"msg":"Invalid value","param":"projectId","location":"body"}]
}

##Screenshots
Case 1: 
Screen Shot 2022-10-02 at 1 45 12 AM

Case 
Screen Shot 2022-10-02 at 1 45 42 AM
2: 

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #162

### CODE REVIEW

**Review by Ari-S-123 — APPROVED** [review]
lgtm

File: src/backend/src/utils/validation.utils.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+
+
+export const intMinZero = (validationObject: any): any => {
```
Comment (anthonybernardi) [inline src/backend/src/utils/validation.utils.ts:3]: validationObject should not be type `any` or return `any` either. check out what the types of the body('asdf') things are in the routes files and put that here

**Review by anthonybernardi — CHANGES_REQUESTED** [review]
great start! Aside from the comment I put, the other thing we want to do is test that the validation still works. You can do this by running the app then sending http requests with an app like postman. Put screenshots of your request and the response in the pr.

Also contributor guideline stuff:
* make sure to tag all commits with the issue # (# sign is important)
* give pr and branch a meaningful name relative to the ticket (branch is fine cuz it's too late, but you can edit pr name)
* make sure this pr is linked to the ticket on the right side where it says development

**Review by anthonybernardi — CHANGES_REQUESTED** [review]
the login endpoint doesn't use any of these validators that you changed. try one like /risks/create -- you'll have to mkae the body match what it's supposed to be in the risks route file. then try it by making the projectId invalid or something (not a >=0 int) and it should fail. if it doesn't fail then validation is now broken

**Review by anthonybernardi — APPROVED** [review]
awesome work! thanks for doing the testing too

**Pull request merged**